### PR TITLE
Sui Network docker compose

### PR DESCRIPTION
**Pull request by johnjmartin**
## Description 

Adds a docker compose for running sui network + loadgen (stress)

Right now the genesis/validator config generation relies on a hacky python script for applying overrides to validator configs. Longer term the ability to do this should probably be moved to the genesis tooling itself

## Test Plan 

tested locally

---
If your changes are not user-facing and do not break anything, you can skip the following section. Otherwise, please briefly describe what has changed under the Release Notes section.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: docker/sui-network/docker-compose.yaml [hunk 1]
```diff
@@ -0,0 +1,134 @@
+version: "3"
+
+services:
```
Comment (arun-koshy) [inline docker/sui-network/docker-compose.yaml:3]: is the extra services here intentional?

File: docker/sui-network/genesis/generate.py [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument(
+        "-g",
+        "--genesis-template",
+        type=str,
+        help="template to use for genesis.yaml generation",
+        required=False,
+    )
+    # parser.add_argument("-n", "--network", type=str, help="network", required=True)
```
Comment (arun-koshy) [inline docker/sui-network/genesis/generate.py:115]: nit: remove commented line

**Review by arun-koshy — APPROVED** [review]

File: docker/sui-network/docker-compose.yaml [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  validator1:
+    networks:
+      sui-network:
+        ipv4_address: 10.0.0.11
+    image: mysten/sui-node:mainnet-v1.19.1
```
Comment (mystenmark) [inline docker/sui-network/docker-compose.yaml:8]: i assume the image tag will become an configurable at some point?

File: docker/sui-network/docker-compose.yaml [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  validator3:
+    networks:
+      sui-network:
+        ipv4_address: 10.0.0.13
+    image: mysten/sui-node:mainnet-v1.19.1
+    container_name: validator3
+    hostname: validator3
+    environment:
+      - RUST_BACKTRACE=1
+      - RUST_LOG=info,sui_core=info,narwhal=info,narwhal-primary::helper=info,jsonrpsee=error
```
Comment (mystenmark) [inline docker/sui-network/docker-compose.yaml:67]: i think we will need `sui_core=debug,sui_network=debug,sui_node=debug,narwhal=debug` on all nodes

File: docker/sui-network/docker-compose.yaml (hunk 3 above)
Comment (johnjmartin) [reply docker/sui-network/docker-compose.yaml]: As this is just an example for running docker-compose, I'm going to leave the image tag pinned on `:mainnet`, which was reintroduced recently: https://github.com/MystenLabs/sui/pull/16588#discussion_r1525173202

**Pull request merged**